node/src/mempool/pool.rs

```diff
@@ -0,0 +1,229 @@
+        }
+
+        // Check minimum fee rate
+        if fee_rate < self.config.min_fee_rate {
+            return Err(MempoolError::FeeTooLow);
+        }
+
+        // Calculate transaction size
+        let tx_size = bincode::serialize(&transaction)
+            .map_err(|_| MempoolError::SerializationError)?
+            .len();
+
+        // Create and insert new entry
+        let entry = MempoolEntry {
+            transaction,
+            timestamp: SystemTime::now(),
+            fee_rate,
+            size: tx_size,
+        };
+
+        self.transactions.insert(tx_hash, entry);
+        Ok(())
+    }
+
+    /// Remove a transaction from the pool
+    pub fn remove_transaction(&self, tx_hash: &[u8; 32]) -> Option<Transaction> {
+        self.transactions.remove(tx_hash).map(|(_, entry)| entry.transaction)
+    }
+
+    /// Get a transaction by its hash
+    pub fn get_transaction(&self, tx_hash: &[u8; 32]) -> Option<Transaction> {
+        self.transactions.get(tx_hash).map(|entry| entry.transaction.clone())
+    }
+
+    /// Clear expired transactions from the pool
+    pub fn clear_expired(&self) -> usize {
+        let now = SystemTime::now();
+        let max_age = Duration::from_secs(self.config.max_age);
+        let mut removed = 0;
+
+        self.transactions.retain(|_, entry| {
+            let age = now.duration_since(entry.timestamp).unwrap_or(Duration::ZERO);
+            if age > max_age {
+                removed += 1;
+                false
+            } else {
+                true
+            }
+        });
+
+        removed
+    }
+
+    /// Check for double-spend attempts
+    pub fn check_double_spend(&self, transaction: &Transaction) -> bool {
+        // Get all input references from the new transaction
+        let new_inputs: Vec<_> = transaction.inputs()
+            .iter()
+            .map(|input| (input.prev_tx_hash(), input.prev_output_index()))
+            .collect();
+
+        // Check if any existing transaction uses the same inputs
+        for entry in self.transactions.iter() {
+            let existing_inputs: Vec<_> = entry.transaction.inputs()
+                .iter()
+                .map(|input| (input.prev_tx_hash(), input.prev_output_index()))
+                .collect();
+
+            // Check for any overlap in inputs
+            for input in &new_inputs {
+                if existing_inputs.contains(input) {
+                    return true;
+                }
+            }
+        }
+
+        false
+    }
+
+    /// Get all transactions sorted by fee rate
+    pub fn get_sorted_transactions(&self) -> Vec<Transaction> {
+        let mut entries: Vec<_> = self.transactions
+            .iter()
+            .map(|ref_multi| {
+                (ref_multi.transaction.clone(), ref_multi.fee_rate)
+            })
+            .collect();
+
+        // Sort by fee rate in descending order
+        entries.sort_by(|a, b| b.1.cmp(&a.1));
```

Comment:
Think you could chain these operations on entries prior to collect, instead of collecting twice over entries.